parsers/rhea/rhea.go

```diff
@@ -0,0 +1,742 @@
+		for _, subclass := range description.Subclass {
+			switch subclass.Resource {
+			case "http://rdf.rhea-db.org/ReactivePart":
+				newCompound, ok := compoundMap[compoundParticipantMap[description.About]]
+				if ok != true {
+					return Rhea{}, errors.New("Could not find " + description.About)
+				}
+				newCompound.ID = description.ID
+				newCompound.CompoundAccession = description.About
+				newCompound.Position = description.Position
+				newCompound.Name = description.Name
+				newCompound.HTMLName = description.HTMLName
+				newCompound.Formula = description.Formula
+				newCompound.Charge = description.Charge
+				newCompound.Chebi = description.Chebi.Resource
+				rhea.Compounds = append(rhea.Compounds, newCompound)
+			}
+		}
+		for _, containsx := range description.ContainsX {
+			if strings.Contains(containsx.XMLName.Local, "contains") {
+				// Get reaction sides
+				// gzip -d -k -c rhea.rdf.gz | grep -o -P '(?<=contains).*(?= rdf)' | tr ' ' '\n' | sort -u | tr '\n' ' '
+				// The exceptions to numeric contains are 2n, N, Nminus1, and Nplus1
+				var newReactionParticipant ReactionParticipant
+				switch containsx.XMLName.Local {
+				case "containsN":
+					newReactionParticipant = ReactionParticipant{
+						ReactionSide: description.About,
+						Contains:     1,
+						ContainsN:    true,
+						Minus:        false,
+						Plus:         false,
+						Accession:    containsx.Content}
+				case "contains2n":
+					newReactionParticipant = ReactionParticipant{
+						ReactionSide: description.About,
+						Contains:     2,
+						ContainsN:    true,
+						Minus:        false,
+						Plus:         false,
+						Accession:    containsx.Content}
+				case "containsNminus1":
+					newReactionParticipant = ReactionParticipant{
+						ReactionSide: description.About,
+						Contains:     1,
+						ContainsN:    true,
+						Minus:        true,
+						Plus:         false,
+						Accession:    containsx.Content}
+				case "containsNplus1":
+					newReactionParticipant = ReactionParticipant{
+						ReactionSide: description.About,
+						Contains:     1,
+						ContainsN:    true,
+						Minus:        false,
+						Plus:         true,
+						Accession:    containsx.Content}
+				default:
+					i, err := strconv.Atoi(containsx.XMLName.Local[8:])
+					if err != nil {
+						return Rhea{}, err
+					}
+					newReactionParticipant = ReactionParticipant{
+						ReactionSide: description.About,
+						Contains:     i,
+						ContainsN:    false,
+						Minus:        false,
+						Plus:         false,
+						Accession:    containsx.Content}
+				}
+				newReactionParticipant.Compound = compoundParticipantMap[description.About]
+				rhea.ReactionParticipants = append(rhea.ReactionParticipants, newReactionParticipant)
+			}
+		}
+	}
+	return rhea, nil
+}
+
+// ReadRhea reads in a a gzip'd Rhea dump (https://www.rhea-db.org/help/download) into bytes.
+func ReadRhea(gzipPath string) ([]byte, error) {
```

Comment:
```suggestion
func ReadGzippedXML(gzipPath string) ([]byte, error) {
```
this function isn't Rhea specific.

Reply:
It should probably go in a `utils` or `io` file.